utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -242,36 +242,83 @@ impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
 }
 
 impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
-    /// Generates a function which calls "properly" the `gr_reservation_send`.
-    fn generate_send_from_reservation(&mut self) {
-        let (reserve_gas_idx, reservation_send_idx) = {
-            let maybe_reserve_gas = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReserveGas))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-            let maybe_reservation_send = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReservationSend))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-
-            match maybe_reserve_gas.zip(maybe_reservation_send) {
-                Some(indexes) => indexes,
-                None => {
-                    log::trace!("Wasm hasn't got either gr_reserve_gas, or gr_reservation_send, or both of them");
-                    return;
-                }
-            }
-        };
+    /// Returns the indexes of invocable sys-calls.
+    fn invocable_sys_calls_indexes(&self, sys_calls: &[SysCallName]) -> Option<Vec<usize>> {
+        let ret: Option<Vec<_>> = sys_calls
+            .iter()
+            .map(|&sys_call| {
+                self.sys_calls_imports
+                    .get(&InvocableSysCall::Loose(sys_call))
+                    .map(|&(_, call_indexes_handle)| call_indexes_handle)
+            })
+            .collect();
+
+        if ret.is_none() {
+            log::trace!(
+                "Wasm should import the following syscalls: {missing_sys_calls}",
+                missing_sys_calls = sys_calls
+                    .iter()
+                    .map(|sys_call| sys_call.to_str())
+                    .collect::<Vec<_>>()
+                    .join(", "),
+            );
+        }
 
-        let invocable_sys_call = InvocableSysCall::Precise(SysCallName::ReservationSend);
-        let send_from_reservation_signature = invocable_sys_call.into_signature();
+        ret
+    }
 
-        let send_from_reservation_func_ty = send_from_reservation_signature.func_type();
+    /// Generates a function which calls "properly" the given sys-call.
+    fn generate_sys_call_pattern(
+        &mut self,
+        sys_call: SysCallName,
+        sys_call_amount: u32,
```

Comment:
this is a constant, I guess. If you want to insert "precise" invocations several times, that must be defined from the config. Unless we have that, I'd remove this argument and would just use `1`.

Reply:
I can set this to 1, but in general it should probably have separate config in the future

Reply:
done in 41651b831611ce73584f0f580835a7db43159fde